### PR TITLE
chore(clusterscope): add unit tests for cluster scope methods

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: Ashutosh Kumar <sonasingh46@gmail.com>

**What type of PR is this?**

Add unit test for scope methods.
Increases the test coverage from 19% to 54.7% 

/kind bug

**What this PR does / why we need it**:

This PR add unit tests for couple of getters methods that has some internal ( e.g `if` / `for` ) logic. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
None

```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/assign @Jont828

File: azure/scope/machine_test.go [hunk 1]
```diff
@@ -49,6 +49,14 @@ func specArrayToString(specs []azure.ResourceSpecGetter) string {
 	return sb.String()
 }
 
+func specToString(spec azure.ResourceSpecGetter) string {
```
[the remark this answers is not in the document]
Comment (sonasingh46) [reply azure/scope/machine_test.go:52]: Sure. I will move it to the other file. There was a similar function there and thought to put closer to that -- this was the only reason. But looks like we can pull all of these helpers functions into a separate file e.g. helpers.go.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**